syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package transport_catalogue_serialize;

message Stop{
	uint32 stop_number_ = 1;
	string stop_name_ = 2;
	double lat = 3;
	double lng = 4;	
}

message Bus{	
	string bus_name_ = 1;
	bool is_circular_ = 2;
	repeated uint32 included_stops_ = 3;
	uint32 bus_number_ = 4;
}

message StopToStopDistance{
	uint32 stop1 = 1;
	uint32 stop2 = 2;
	double distance = 3;
}

message TransportCatalogue{
	repeated Stop stops_ = 1;
	repeated Bus buses_ = 2;
	repeated StopToStopDistance distances_= 3;
	RenderSettings render_settings_ = 4;
	TransportCatalogueGraph graph_ = 5;
	RouterSettings router_settings_ = 6;
}